sstables/metadata_collector.cc

```diff
@@ -61,4 +61,28 @@ void metadata_collector::update_first_and_last_clustering_positions(position_in_
     _last_clustering_pos = pos;
 }
 
+clustering_position_metadata metadata_collector::make_clustering_position_metadata() const {
+    auto to_exclusive = [] (bool is_first_pos, bound_weight w) -> std::optional<uint8_t> {
+        switch (w) {
+        case bound_weight::equal:
+            return std::nullopt;
+        case bound_weight::before_all_prefixed:
+            return (is_first_pos) ? false /*bound_kind::incl_start*/ : true /*bound_kind::excl_end*/;
+        case bound_weight::after_all_prefixed:
+            return (is_first_pos) ? true /*bound_kind::excl_start*/ : false /*bound_kind::incl_end*/;
```

Comment:
Why not just encode `position_in_partition` "verbatim" here? This is our own component, no need to use the clunky `bound_kind` here. We can just write the clustering elements + a weight, as-is in the `position_in_partition`.

Reply:
I asked myself if the concept of position_in_partition should be persisted to disk. Ideally, metadata written to disk is more generic. But you have a good point, this is scylla-specific metadata, not supposed to be used elsewhere.